analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/BranchingProcessor.cs

```diff
@@ -52,8 +52,8 @@ protected override ProgramState[] Process(SymbolicContext context, T operation)
                 ? beforeLearningState.ToArray()   // We can't learn anything, just move on
                 : new[]
                 {
-                    positive.SetOperationConstraint(context.Operation, BoolConstraint.True),
-                    negative.SetOperationConstraint(context.Operation, BoolConstraint.False)
+                    positive.SetOperationConstraint(context.Operation, ObjectConstraint.NotNull).SetOperationConstraint(context.Operation, BoolConstraint.True),
+                    negative.SetOperationConstraint(context.Operation, ObjectConstraint.NotNull).SetOperationConstraint(context.Operation, BoolConstraint.False)
```

Comment:
I'm not sure about this change. It basically sets the NotNull constraint for `IsNull` operations for both branches.

Reply:
It also adds `NotNull` for results of like `==`, `is string`.

And in all cases, they for the operation to take the next branching decision. And nobody else has really access to it, as it will get throw away.

I'm fine with not doing this change. Unless it fixes a concrete problem that will appear.

Reply:
...unless it fixes the failed assertion that pops up on running ITs